src/main/python/index.ts

```diff
@@ -28,23 +28,23 @@ export async function checkPythonInstallation(
   }
   if (existsSync(interpreterCachePath)) {
     const interpreter = readFileSync(interpreterCachePath).toString("utf-8");
-    if (existsSync(interpreter)) {
-      const foundInList = global.pythonInterpreters.find(
+    const matchedVersion = await PythonManager.checkVersion(interpreter, {
+      major: 3,
+      minor: 11,
+    });
+    if (matchedVersion) {
+      const foundInterpreterInList = global.pythonInterpreters.find(
         (i) => i.path === interpreter,
       );
-      if (foundInList) {
-        global.pythonInterpreters = global.pythonInterpreters.map((i) =>
-          i.path === interpreter
-            ? {
-                ...i,
-                default: true,
-              }
-            : i,
-        );
+      if (foundInterpreterInList) {
+        global.pythonInterpreters = global.pythonInterpreters.map((i) => ({
+          ...i,
+          default: i.path === interpreter ? true : false,
+        }));
```

Comment:
seems like the `find` and `map` here can just be combined into a single `map`?

Reply:
@itsjoeoui I think map and find can not be combined here as Python interpreter retrieved from cache file can possibly not discovered by Python manager. In that case we need to check and push it to pythonInterpreter list.